source/common/router/router.cc

```diff
@@ -502,7 +532,15 @@ void Filter::onRequestComplete() {
   downstream_request_complete_time_ = dispatcher.timeSource().monotonicTime();
 
   // Possible that we got an immediate reset.
-  if (upstream_request_) {
+  bool any_upstreams = false;
+  for (unsigned long i = 0; i < upstream_requests_.size(); i++) {
+    if (upstream_requests_[i]) {
```

Comment:
Do you populate this with nulls at some point? If not i would imagine you can just check that this is not empty?